### PR TITLE
feat(transfer-request): refactor transfer request page

### PR DESCRIPTION
**Pull request by marvelm57**
Changes:
1. Buat komponen verifikasi OTP yang bisa digunakan di mana saja dan dapat dikustomisasi isinya.
2. Memisahkan logika bisnis, handle form, dan fetch data dari tampilan UI pada halaman transfer request.
3. Mengikuti best practice dan prinsip SOLID untuk meningkatkan maintainability dan testability.

## Bukti coverage
- **Transfer Request**
Screenshot 2025-04-21 at 23 27 01

- **OTP Verification**
Screenshot 2025-04-21 at 23 27 52

## Summary by CodeRabbit

- **New Features**
  - Introduced OTP verification components, including a reusable OTP input form and verification card, for enhanced document transfer security.
  - Added a component to display document information, such as current owner email and document name, with privacy protection.
  - Implemented a custom hook to manage transfer request logic and state.

- **Refactor**
  - Simplified the transfer request module by consolidating logic into a custom hook and using new OTP verification components for a cleaner UI.

- **Tests**
  - Added comprehensive test suites for OTP verification and transfer document components, covering various user interaction scenarios and edge cases.

- **Chores**
  - Removed unused comment directives to improve code cleanliness.

### CODE REVIEW

**Comment by marvelm57** [comment]
@coderabbitai review

File: src/components/modules/shared/OTPVerification/DocumentInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}: Readonly<DocumentInfoProps>) {
+  if (!currentOwnerEmail && !documentName) return null;
+
+  return (
+    <div className="space-y-1">
+      {currentOwnerEmail && (
+        <p className="text-xs">
+          From:{' '}
+          <span className="font-semibold">
+            {encryptEmail(currentOwnerEmail)}
```
Comment (CyberSleeper) [inline src/components/modules/shared/OTPVerification/DocumentInfo.tsx:20]: Oke nice uda make existing function

File: src/components/modules/shared/OTPVerification/OTPInputForm.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          </InputOTPGroup>
+        </InputOTP>
+      </div>
+
+      {responseMessage ? (
+        <p className="text-xs text-primary-1">{responseMessage}</p>
+      ) : (
+        onResend && (
+          <div className="flex items-center gap-1">
+            <p className="text-xs text-primary-1">Didn&apos;t get the code? </p>
```
Comment (CyberSleeper) [inline src/components/modules/shared/OTPVerification/OTPInputForm.tsx:55]: KERENNN MARVELLLL
Ini koe pasti kena eror lint pas build makanya ganti jadi apos
Makasii ya uda nge-build dlu sebelum PR 🤩🤩

File: src/components/modules/transfer-request/index.tsx [hunk 3]
```diff
@@ -1,103 +1,40 @@
...
-              variant="default"
-              className="w-full"
-              onClick={handleSubmit}
-              disabled={otp.length !== 6 || isLoadingClaimDocument}
-            >
-              {isLoadingClaimDocument ? 'Verifying...' : 'Verify'}
-            </Button>
-          </div>
-        </div>
+        <OTPVerificationCard
```
Comment (CyberSleeper) [inline src/components/modules/transfer-request/index.tsx:28]: Gokilll uda dibikin modular gini

File: src/components/modules/transfer-request/hooks/use-transfer-request.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { useParams } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import { UseMetadata } from '../../metadata/hooks/use-metadata';
+import { useClaimDocument } from './use-claim-document';
+
+export function useTransferRequest() {
```
Comment (CyberSleeper) [inline src/components/modules/transfer-request/hooks/use-transfer-request.ts:6]: Oke baru kepikiran bisa diginiin. Setauku lebih baik klo yg kecil2 gini ttep ditaruh di file yg make dia sih. Tapi mungkin kita bisa ngobrolini ttg ini besok.
Anw kerenn Marvell

**Review by CyberSleeper — APPROVED** [review]
Keren bangett sih ini
Makasii Marvell
